expression/function_traits_test.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expression
+
+import (
+	. "github.com/pingcap/check"
+	"github.com/pingcap/tidb/ast"
+	"github.com/pingcap/tidb/util/testleak"
+)
+
+func (s *testEvaluatorSuite) TestUnfoldableFuncs(c *C) {
+	defer testleak.AfterTest(c)()
+	_, ok := unFoldableFunctions[ast.Sysdate]
```

Comment:
When the issue was fixed in the first time, we also added a unit test but it did not pervent from change. Maybe it is better to add explain test or add some comments so that we know why it is unfoldable.

Reply:
A explain test is added, PTAL.